main/src/ca/uwaterloo/flix/api/lsp/Indexer.scala

```diff
@@ -252,6 +256,17 @@ object Indexer {
     case Expr.LetRec(sym, _, _, exp1, exp2, _, _, _) =>
       Index.occurrenceOf(sym, exp1.tpe) ++ visitExp(exp1) ++ visitExp(exp2) ++ Index.occurrenceOf(exp0)
 
+    case Expr.LocalDef(_, sym, fparams, declaredType, declaredEff, exp1, exp2, _, _, _) =>
+      Index.all(
+        traverse(fparams)(visitFormalParam),
+        visitType(declaredType),
+        visitType(declaredEff),
+        Index.occurrenceOf(sym, exp1.tpe),
```

Comment:
Doesn't this imply that for `def example(x: Int32): Int32 = ???`, `example` would be given type `Int32`? I think you should use the `itpe` or whatever its called

Reply:
By given he means shown on hover in VSCode

Reply:
I think it might be good to build a JAR and do a bit of hover + renaming to see that everything works as expected.